### DISCUSSION TITLE
Request: Personal pseudo org-wide Secret configuration

### DISCUSSION THREAD

**Discussion by mcandre** [community/community#84075] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi,

I have a rather large number of repositories which I need to manage the same GitHub Actions Secret for.

Rather than manually setting up each individual repository with the secret, I would prefer to setup the secret exactly once, to reuse across all my personal pseudo-org's repositories.

When I modify the secret over time, per good security practice, then I don't want to have to repeat the change across *hundreds* of repositories. I want to be able to do this exactly once, at my personal account level.

No, I don't want to convert my personal account to a formal GitHub organization. That would lock me out of my account.

**Comment by codesmith-emmy** [comment]
Hello, you can set an organisation repo and then scope the secret to one or more repositories.

You could also setup an Azure Key Vault or similar and run a script to read the secrets and set them as a secret for the action job.

Please visit [GitHub Organization Secrets](https://www.thegeekdiary.com/gh-secret-set-create-or-update-github-secrets-from-the-command-line) for more information.

**Comment by mcandre** [comment]
From a UX perspective, that still looks like a lot of work compared to
creating secrets at the personal account level.

On Wed, Jan 3, 2024 at 7:15 PM emmanuel ***@***.***> wrote:

> Hello, you can set an organisation repo and then scope the secret to one
> or more repositories.
>
> You could also setup an Azure Key Vault or similar and run a script to
> read the secrets and set them as a secret for the action job.
>
> Please visit GitHub Organization Secrets
> <https://www.thegeekdiary.com/gh-secret-set-create-or-update-github-secrets-from-the-command-line>
> for more information.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/84075#discussioncomment-8008307>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAABJRBCAUNMDC36YPR3J5LYMX7DLAVCNFSM6AAAAABBMB2UZWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DAMBYGMYDO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Cheers,
Andrew
